browser/new_tab/new_tab_shows_options.cc

```diff
@@ -0,0 +1,83 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/new_tab/new_tab_shows_options.h"
+
+#include <utility>
+
+#include "base/values.h"
+#include "brave/browser/profiles/profile_util.h"
+#include "brave/common/pref_names.h"
+#include "brave/grit/brave_generated_resources.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/ui/webui/ntp/new_tab_ui.h"
+#include "chrome/common/pref_names.h"
+#include "chrome/common/webui_url_constants.h"
+#include "components/prefs/pref_service.h"
+#include "ui/base/l10n/l10n_util.h"
+#include "url/url_constants.h"
+
+namespace brave {
+
+GURL GetNewTabPageURL(Profile* profile) {
+  DCHECK(profile);
+
+  if (!brave::IsRegularProfile(profile))
+    return GURL();
+
+  auto* prefs = profile->GetPrefs();
+
+  int option = prefs->GetInteger(kNewTabPageShowsOptions);
+  if (option == brave::NewTabPageShowsOptions::HOMEPAGE) {
+    if (prefs->GetBoolean(prefs::kHomePageIsNewTabPage))
+      return GURL();
+    return GURL(prefs->GetString(prefs::kHomePage));
+  } else if (option == brave::NewTabPageShowsOptions::BLANKPAGE) {
+    return GURL(url::kAboutBlankURL);
```

Comment:
As commented, how about we use regular existing NTP URL for `BLANKPAGE` option and just have it render a blank page that we can control bg of and anything else we may want to do with a "blank" page?

Reply:
@petemill good idea! 👍